src/Misc/externals.sh

```diff
@@ -32,13 +32,29 @@ function checkRC() {
 function acquireExternalTool() {
     local download_source=$1 # E.g. https://vstsagenttools.blob.core.windows.net/tools/pdbstr/1/pdbstr.zip
     local target_dir="$LAYOUT_DIR/externals/$2" # E.g. $LAYOUT_DIR/externals/pdbstr
-    local fix_nested_dir=$3 # Flag that indicates whether to move nested contents up one directory. E.g. TEE-CLC-14.0.4.zip
-                            # directly contains only a nested directory TEE-CLC-14.0.4. When this flag is set, the contents
-                            # of the nested TEE-CLC-14.0.4 directory are moved up one directory, and then the empty directory
-                            # TEE-CLC-14.0.4 is removed.
+    local fix_nested_dir # Flag that indicates whether to move nested contents up one directory. E.g. TEE-CLC-14.0.4.zip
+                         # directly contains only a nested directory TEE-CLC-14.0.4. When this flag is set, the contents
+                         # of the nested TEE-CLC-14.0.4 directory are moved up one directory, and then the empty directory
+                         # TEE-CLC-14.0.4 is removed.
+    local download_name  # Override the downloaded file name if not obvious in $download_source
+
+    while [[ $# -gt 0 ]]; do
```

Comment:
To the review, I felt rather than "magic parameter positions", being more explicit in the command lines below was more obvious to future editors going forward. This is a pretty common pattern for parsing parameters without a dependency on `getopts` (which only supports short params).